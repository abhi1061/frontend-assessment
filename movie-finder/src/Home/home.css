.home-container {
    display: flex;
    flex-direction: column;
}
.banner {
    height: 7vh;
    background-color: #335267;
    border-bottom-right-radius: 0;
    border-bottom-left-radius: 0;
}

.body-card {
    border-top-left-radius: 0;
    border-top-right-radius: 0;
}

.btn-outline-secondary:active:focus {
  color: #ffffff; 
  background-color: #18344A; 
  border-color: #494F57;
}

.loader {
    display: flex;
    flex: 1;
    flex-direction: row;
    justify-content: center;
    align-items: center;
}