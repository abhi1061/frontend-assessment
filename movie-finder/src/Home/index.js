import React, { useState, useEffect } from 'react';
import _ from 'lodash';

import useInfiniteScroll from './useInfiniteScroll';
import { getSearchResult } from '../Actions';
import './home.css';
import SearchBox from '../SearchBox';
import CategorySelection from '../Category';
import List from '../List';

export default function Home(props) {
  const [state, setState] = useState([]);
  const [page, setPage] = useState(1);
  const [query, setQuery] = useState('e'); // Initial query param as wery is required parameter
  const [category, setCategory] = useState('all');

  const [isFetching, setIsFetching] = useInfiniteScroll(fetchMoreListItems);

  async function fetchData(query, page) {
    if (query === '') {
      return;
    }
    let response = await getSearchResult(query, page);
    if (isFetching) {
      setState((prevState) => [...prevState, ...response.results]);
    } else {
      setState(response.results);
    }
  }

  function fetchMoreListItems() {
    setTimeout(() => {
      setPage(page + 1);
      setIsFetching(false);
    }, 2000);
  }

  useEffect(() => {
    fetchData(query, page);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [query, page]);

  function filterData() {
    if (category !== 'all') {
      return _.filter(state, ['media_type', category]);
    }
    return state;
  }

  return (
    <div className="home-container">
      <div className="card shadow banner"></div>
      <div className="card shadow p-4 body-card">
        <SearchBox setQuery={setQuery} />
        <CategorySelection category={category} setCategory={setCategory} />
        {state.length ? (
          <List id="list" data={filterData()} />
        ) : (
          <span>No Data Found</span>
        )}
      </div>
      {isFetching ? (
        <div className="loader mt-3">
          <i className="fa fa-lg fa-spinner fa-pulse"></i>
        </div>
      ) : null}
    </div>
  );
}
