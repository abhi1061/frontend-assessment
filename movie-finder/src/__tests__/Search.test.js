import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Home from '../Home';

Enzyme.configure({ adapter: new Adapter() });

describe('<Home /> with no props', () => {
  let container;
  let setQuery = jest.fn();
  beforeEach(() => {
    container = Enzyme.mount(<Home setQuery={setQuery} />);
  });

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should have a search field', () => {
    expect(container.find('#search').length).toEqual(1);
  });
});
