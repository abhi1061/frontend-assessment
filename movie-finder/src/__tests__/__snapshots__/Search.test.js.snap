// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> with no props should match the snapshot 1`] = `"<div class=\\"home-container\\"><div class=\\"card shadow banner\\"><img alt=\\"Logo\\" src=\\"Logo.svg\\"></div><div class=\\"card shadow p-4 body-card\\"><div class=\\"input-group mb-3\\"><div class=\\"input-group-prepend\\"><span class=\\"input-group-text pre-noback\\"><i class=\\"fa fa-xs fa-search\\"></i></span></div><input id=\\"search\\" type=\\"text\\" class=\\"form-control\\" placeholder=\\"Search for movies, tv shows, people...\\" aria-label=\\"Search for movies, tv shows, people...\\" aria-describedby=\\"basic-addon2\\"><div class=\\"input-group-append\\"><button class=\\"btn btn-outline-secondary searchBtn\\" type=\\"button\\">Search</button></div></div><div class=\\"mb-3\\"><button type=\\"button\\" class=\\"btn btn-outline-secondary btn-sm mr-3 pr-3 pl-3 selected\\">All</button><button type=\\"button\\" class=\\"btn btn-outline-secondary btn-sm mr-3 pr-3 pl-3\\">Movies</button><button type=\\"button\\" class=\\"btn btn-outline-secondary btn-sm mr-3 pr-3 pl-3\\">TV Shows</button><button type=\\"button\\" class=\\"btn btn-outline-secondary btn-sm mr-3 pr-3 pl-3\\">People</button></div><h6>No Data Found</h6></div></div>"`;
